cluster/src/main/java/org/apache/iotdb/cluster/query/ClusterPlanExecutor.java

```diff
@@ -81,10 +97,48 @@ public QueryDataSet processQuery(PhysicalPlan queryPlan, QueryContext context)
   }
 
   @Override
-  protected List<String> getNodesList(String schemaPattern, int level) {
-    // TODO-Cluster: enable meta queries
-    throw new UnsupportedOperationException("Not implemented");
-    //return metaGroupMember.getNodeList(schemaPattern, level);
+  protected List<String> getNodesList(String schemaPattern, int level)
+      throws IOException, TException, InterruptedException, MetadataException {
+    Set<String> nodeListSet = new HashSet<>(
+        MManager.getInstance().getNodesList(schemaPattern, level));
+
+    GetNodesListHandler handler = new GetNodesListHandler();
+
+    ExecutorService pool = new ScheduledThreadPoolExecutor(THREAD_POOL_SIZE);
+
+    for (Node node : metaGroupMember.getAllNodes()) {
+      if (node.equals(metaGroupMember.getThisNode())) {
+        continue;
+      }
+      pool.submit(() -> {
+        DataClient client = null;
+        try {
+          client = metaGroupMember.getDataClient(node);
+        } catch (IOException e) {
+          logger.error("Failed to connect to node: {}", node, e);
+        }
+        AtomicReference<List<String>> response = new AtomicReference<>(null);
+        handler.setResponse(response);
+        handler.setContact(node);
```

Comment:
The same handler is used in different threads and its response and contact are set concurrently, so I am afraid that only the result from one node is returned because there is only one response in the handler.

And the answer is still correct because the default replica number and node number are both 3, so timeseries are fully replicated in each node and even if you only query one node, it will have all timeseries. Thus I suggest you test in a different configuration, like 4-node-3-replica or 3-node-2-replica.

Reply:
Yes, there's a problem. Thanks for your helpful advice. The initialization of  `response` now is inside each thread.

Reply:
Yes, there's a problem. Thanks for your reminding. The initialization of handler should be inside each thread.